data_pipeline/twitter/data_from_twitter_rest_api.py

```diff
@@ -0,0 +1,302 @@
+import time
+import logging
+from typing import Iterable
+
+import tweepy
+from google.cloud.bigquery.client import Client
+from tweepy import API, User
+from tweepy.cursor import BaseIterator
+
+from data_pipeline.twitter.process_tweet_response import (
+    extract_tweet_from_twitter_response,
+    modify_by_search_term_occurrence_location_in_response
+)
+from data_pipeline.twitter.twitter_bq_util import (
+    batch_load_iter_record_to_bq, batch_load_multi_iter_record_to_bq
+)
+from data_pipeline.twitter.twitter_config import (
+    TwitterDataPipelineConfig, TweetType,
+    TwitterPipelineModuleConstants, ETL_TIMESTAMP_FORMAT
+)
+from data_pipeline.utils.data_store.bq_data_service import get_bq_table
+from data_pipeline.utils.pipeline_file_io import WriteIterRecordsInBQConfig
+
+from data_pipeline.utils.record_processing import (
+    add_provenance_to_record, standardize_record_keys
+)
+
+LOGGER = logging.getLogger(__name__)
+
+
+def _limit_handled(cursor: BaseIterator):
```

Comment:
Have you tried using the tweepy API with `wait_on_rate_limit=True`? In theory that should already handle it? There is also `wait_on_rate_limit_notify` to show some sort of message.
(I haven't tried either)

(I was checking because I wasn't sure where the 15 minutes sleep time came from)